internal/configs/version2/nginx.virtualserver.tmpl

```diff
@@ -8,8 +8,10 @@ upstream {{ $u.Name }} {
     server {{ $s.Address }} max_fails={{ $s.MaxFails }} fail_timeout={{ $s.FailTimeout }};
     {{ end }}
 
-    {{ if $.Keepalive }}
-    keepalive {{ $.Keepalive }};
+    {{ if $u.Keepalive }}
```

Comment:
We need to set the keeaplive directive if u.Keepalive > 0. Note that `0` is invalid value here. 
As mentioned https://github.com/nginxinc/kubernetes-ingress/pull/617/files#r301005434, you need to set the connection header to `""` if keepalive is enabled.